src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -108,9 +108,24 @@ public void sendPackets(
             {
                 if (citizen.isDirty() || hasNewSubscribers)
                 {
-                    subscribers.stream()
-                            .filter(player -> citizen.isDirty() || !oldSubscribers.contains(player))
-                            .forEach(player -> MineColonies.getNetwork().sendTo(new ColonyViewCitizenViewMessage(colony, citizen), player));
+                    if (citizen.getCitizenEntity().isPresent())
+                    {
+                        final List<EntityCitizen> list = colony.getWorld()
+                                .getEntities(EntityCitizen.class,
+                                        entityCitizen -> entityCitizen.getColony().getID() == colony.getID() && entityCitizen.getCitizenData().getId() == citizen.getId());
+
+                        if (!list.isEmpty() && citizen.getCitizenEntity().get().getEntityId() != list.get(0).getEntityId())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3655)